salt/_auth/kubernetes_rbac.py

```diff
@@ -147,20 +147,39 @@ def _load_kubeconfig(opts):
     return kubeconfig
 
 
-def auth(username, token, token_type, **kwargs):
+def auth(username, password=None, token=None, **kwargs):
     log.info('Authentication request for "%s"', username)
 
-    handler = AUTH_HANDLERS.get(token_type.lower(), None)
-    if not handler:
-        log.warning('Unknown auth token_type: %s', token_type)
+    if token and password:
+        log.warning(
+            'Invalid authentication request cannot provide "token" and '
+            '"password" at the same time'
+        )
+        return False
+    if not token and (not password or not username):
```

Comment:
Don't change it, but in the future, you can make this boolean expression more readable with:

```python
if not (token or password and username):

# Or with parentheses
if not (token or (password and username)):
```

Reply:
An underrated pattern:
```python
if not any([
        token,
        username and password,
        ]):
    blah()
```

Reply:
Like this one, it's even more readable :)

Reply:
While strictly speaking less efficient (so could make a difference in 'hot' loops and such), `any` and `all` often make multiple `or`'ed or `and`'ed boolean expressions easier to grasp.